core/src/main/java/org/apache/metamodel/schema/typing/ColumnTypingContextImpl.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.metamodel.schema.typing;
+
+import org.apache.metamodel.schema.ColumnType;
+import org.apache.metamodel.schema.Table;
+
+
+/**
+ * An implementation of {@link ColumnTypingContext} that holds necessary context about the column being configured.
+ */
+public class ColumnTypingContextImpl implements ColumnTypingContext {
+
+    private final int columnIndex;
+
+    private final Table table;
+
+
+    /**
+     * Creates a context to conifgure a column for a specific table.
+     * @param table               The table that contains the column
+     * @param columnIndex         the index in the table of the column being configured.
+     */
+    public ColumnTypingContextImpl( Table table, int columnIndex ) {
```

Comment:
I will go back and double check the space padding in the rest of the files. My work uses different formatting rules and I'm not used to handling multiple rulesets in my IDE.

I agree with making parameters final when possible. Does this also apply to parameters of lambda functions? for example an instance of the ColumnTypingSession after adding the default implementation to close:
```
        return (final ColumnTypingContext ctx) -> {
            if ( iterator.hasNext() ) {
                return iterator.next();
            }
            return null;
        };
```
or 
```
        return ctx -> {
            if ( iterator.hasNext() ) {
                return iterator.next();
            }
            return null;
        };
```

Personally I treat parameters as final whether they are marked as such or not, and I don't see any final lambda parameters in the project so far.

Reply:
You make a valid point, because with the second example you're able to assign a new Object (if you were able to instantiate `ColumnTypingContext`) to the `ctx` parameter. Even though I wouldn't consider doing that.

It's just that in practice I always see the second example (the one you implemented), so for lambdas let's not start adding finals (and types) to the formal parameters now.